### ISSUE TITLE
(yaml) Add support for inline sequences and mappings

### ISSUE THREAD

**Issue by pplantinga** [highlightjs/highlight.js#2498]
**Describe the issue**
Current highlighting does not support inline sequences and mappings.

**Which language seems to have the issue?**
YAML

**Sample Code to Reproduce**
https://jsfiddle.net/sda50gr8/

**Expected behavior**
![sequence-mapping](https://user-images.githubusercontent.com/4369106/79674618-dc652780-81b2-11ea-8d73-a284b4ce5485.png)

**Comment by yyyc514** [comment]
Can these be recursive also?

**Comment by pplantinga** [comment]
Yes, these can be nested. I've started working on this, but the simple implementation (just add a rule for all brackets and commas) auto-detected a "clean" file as yaml.

**Comment by yyyc514** [comment]
Wait. What is the goal here again?  I’m not certain it’s clear from the snap.

**Comment by pplantinga** [comment]
I think top priority should be given to making sure numbers are not highlighted as strings (e.g. `{1` or `2.0,` or `3]` would all be highlighted as strings). A secondary concern might be highlighting braces, brackets, and commas as a separate type than strings.

**Comment by yyyc514** [comment]
> A secondary concern might be highlighting braces, brackets, and commas as a separate type than strings.

Traditionally we don't highlight symbols.

You might look into the JSON grammar for inspiration.
